build_tools/cmake/build_e2e_test_artifacts.sh

```diff
@@ -101,7 +101,8 @@ FLAG_DUMP="${E2E_TEST_ARTIFACTS_DIR}/benchmark-flag-dump.txt"
 ./build_tools/benchmarks/export_benchmark_config.py \
   execution \
   --benchmark_presets="${EXECUTION_PRESETS}" \
-  --output="${EXECUTION_CONFIG}"
+  --output="${EXECUTION_CONFIG}" \
+  --shard_count="a2-highgpu-1g=2" # Just an experiment for now
```

Comment:
Change here will affect all workflows using this script. We might want to make it a parameter at workflow level (`build_e2e_test_artifacts.yml`) so that it is configurable.

It looks like we don't have a lot of benefit to shard in presubmit because currently x86_64 and CUDA only take 11mins. But it will be valuable to enable in `benchmark_large.yml`, which takes a long time to run: https://github.com/openxla/iree/actions/runs/5220190187

Reply:
Yes, the plan was always to plumb it through to the outermost workflow. Not sure though how we would prefer to do that.

But it sounds like this would work:

1. Make the shart count string an input parameter of the `build_e2e_test_artifacts` workflow
2. Set this parameter to `default=1` in the presubmit/postsubmit (CI.yml) case and to `a2-highgpu-1g=2,default=1` in the benchmark large case.
3. Ideally make this overridable in benchmark large case. (I vaguely recall that workflow_dispatch trigger allow overwriting some parameters.)

Does that make sense to you?